clang/lib/CIR/CodeGen/CIRGenExprComplex.cpp

```diff
@@ -174,6 +185,58 @@ mlir::Value ComplexExprEmitter::emitCast(CastKind ck, Expr *op,
   return {};
 }
 
+mlir::Value ComplexExprEmitter::VisitUnaryPlus(const UnaryOperator *e,
+                                               QualType promotionType) {
+  QualType promotionTy = promotionType.isNull()
+                             ? getPromotionType(e->getSubExpr()->getType())
+                             : promotionType;
+  mlir::Value result = VisitPlus(e, promotionTy);
+  if (!promotionTy.isNull()) {
+    cgf.cgm.errorNYI("ComplexExprEmitter::VisitUnaryPlus emitUnPromotedValue");
+    return {};
+  }
+
+  return result;
+}
+
+mlir::Value ComplexExprEmitter::VisitPlus(const UnaryOperator *e,
```

Comment:
Can `VisitPlus` and `VisitMinus` be combined into a single `emitUnaryPlusOrMinus` as they are in `ScalarExprEmitter`?

Reply:
I still think these can be combined, and I don't think they are actually called by the Visitor pattern, so it would be better to rename them. If I'm wrong about them not being called by the Visitor they should stay.

Reply:
I will try to combine them into one function similar to what we did with `VisitPrePostIncDec` and check the visitor also